app/src/components/adminConsole/currentCup/CreateCupButton.tsx

```diff
@@ -9,17 +12,53 @@ const CreateCupButton: React.FunctionComponent = (props: {
     isOpen: Boolean;
 }) => {
     const [status, setStatus] = useState({ status: IDLE });
-    const createCup = () => {
-        console.log("cup created");
+    const [newCupName, setNewCupName] = useState<String>("");
+    const [allCupNames, setAllCupNames] = useState<String[]>([]);
+
+    useEffect(() => {
+        GetCupNames()
+            .then((res: String[]) => {
+                setAllCupNames(res);
+            })
+            .catch((e) => console.log(e));
+    }, [GetCupNames, setAllCupNames]);
+
+    const handleCreateClick = () => {
+        setStatus({ status: LOADING });
+        const newCup: Cup = {
+            isPublished: false,
+            isOpen: false,
+            timeCreated: new Date(),
+            name: newCupName,
+        };
+        try {
+            createNewCup(newCup);
+            setNewCupName("");
+            //TODO: change isCup
+        } catch (error) {
+            console.log(error.toString());
+            console.log("error in firebase");
+        }
+        setStatus({ status: IDLE });
     };
+
+    const handleNCNChange = (event) => {
+        setNewCupName(event.target.value);
+    };
+
     if (!props.isCup) {
         return (
             <div>
+                <input value={newCupName} onChange={handleNCNChange} />
                 <button
                     type="button"
                     className="btn-createCup"
-                    onClick={() => createCup()}
-                    disabled={status.status === LOADING}
+                    onClick={() => handleCreateClick()}
+                    disabled={
+                        status.status === LOADING ||
+                        newCupName.length == 0 ||
+                        allCupNames.includes(newCupName)
```

Comment:
I'd be inclined to _not_ include this for the moment. There's no UI to show us the other cups, so it'll just be confusing.

Reply:
I've kept in the 'length=0' disabling, the other cups is gone